celery/canvas.py

```diff
@@ -663,8 +663,16 @@ def run(self, args=None, kwargs=None, group_id=None, chord=None,
             first_task = tasks.pop()
             options = _prepare_chain_from_options(options, tasks, use_link)
 
-            first_task.apply_async(**options)
-            return results[0]
+            real_result = first_task.apply_async(**options)
```

Comment:
The term `real result` is a little bit unclear.
Can we find a better name for it?

Reply:
Hmm, perhaps `concrete_result` or `result_from_apply`? The comment block below this would explain why we'd be using `result_from_apply` adequately.